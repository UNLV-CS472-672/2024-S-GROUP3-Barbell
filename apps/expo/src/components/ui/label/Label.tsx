import { ColorValue, Text, View } from 'react-native'

interface LabelProps {
  text: string
  textColor: ColorValue
  backgroundColor: ColorValue
  rounded?: 'none' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl'
  fontSize?: number
}

export default function Label({
  text,
  textColor,
  backgroundColor,
  rounded = 'lg',
  fontSize = 20,
}: LabelProps) {
  return (
    <View
      style={{
        alignSelf: 'flex-start',
        backgroundColor: backgroundColor,
      }}
      className={`rounded-${rounded} px-2 py-1`}
    >
      <Text
        testID='text-test'
        className={`text-[${fontSize}px] font-bold`}
        style={{ color: textColor }}
        numberOfLines={1}
      >
        {text}
      </Text>
    </View>
  )
}
