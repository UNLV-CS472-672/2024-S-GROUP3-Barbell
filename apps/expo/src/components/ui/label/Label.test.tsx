import React from 'react'

import { render, screen } from '@testing-library/react-native'

import Label from './Label'

test('Label', async () => {
  const expectedLabelText = 'Label'
  const expectedBackgroundColor = '#FFFFFF'
  const expectedTextColor = '#000000'

  render(<Label text={expectedLabelText} backgroundColor={expectedBackgroundColor} textColor={expectedTextColor} />)

  const actualLabelText = await screen.findByTestId('text-test')

  expect(actualLabelText).toHaveTextContent(expectedLabelText)
  expect(screen.toJSON()).toMatchSnapshot()
})
