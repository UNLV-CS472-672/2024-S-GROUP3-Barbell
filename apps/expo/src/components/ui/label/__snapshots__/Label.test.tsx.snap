// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label 1`] = `
<View
  className="rounded-lg px-2 py-1"
  style={
    {
      "alignSelf": "flex-start",
      "backgroundColor": "#FFFFFF",
    }
  }
>
  <Text
    className="text-[20px] font-bold"
    numberOfLines={1}
    style={
      {
        "color": "#000000",
      }
    }
    testID="text-test"
  >
    Label
  </Text>
</View>
`;
